core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -96,6 +114,7 @@ private[spark] object JsonProtocol {
         executorMetricsUpdateToJson(metricsUpdate)
       case blockUpdated: SparkListenerBlockUpdated =>
         throw new MatchError(blockUpdated)  // TODO(ekl) implement this
+      case _ => parse(write(event))
```

Comment:
if some more generic handling of json marshalling is in there (nice), why is `SparkListenerBlockUpdated` being excluded? It should just be one of the types marshalled